### ISSUE TITLE
Building to ESP32

### ISSUE THREAD

**Issue by hunabapr** [vdeschwb/esp32-signal-generator#4]
Hi,
I've tried both Arduino and VS code to upload this to the esp32 following the process to upload the data to spiff first but the spiff data does appears to have a problem although the upload shows to have been successful.
When connecting to configure the esp32 in AP mode I can't, I get the "File Not Found" message if using http://192.168.4.1/config.html but if using http://192.168.4.1/config I get {"ssid":"","local_ip":"0.0.0.0","gateway":"0.0.0.0","subnet":"0.0.0.0"}

Any ideas?
Thanks in advance  Paul

**Comment by DarS007** [comment]
Yeah, I faced the same issue. The _config.html_ file is missing. It is not included in the Github repository, therefore the device configuration page (http://192.168.4.1/config.html) cannot work. Can we ask to update the repo please?  
And thanks for the great code! I was searching for a simple PWM generator, and found much more here.
Regards,
-DarS

**Comment by DarS007** [comment]
Wow, interesting finding...! I accidentally discovered that the web interface looks differently on mobile (Android) and PC (MS Windows). When browsing on PC (Firefox), the web interface has two buttons **Configuration** and **About** that are **missing** on Android browsers (checked with Chrome and Firefox Focus).
Therefore the initial (first) configuration of the device shall be made with PC, where you have the access to **Configuration** page. Set the WiFi credentials properly there and the ESP32 will connect to your network. You then can proceed with mobile browsers if you wish.
-DarS

**Comment by Anthocyanina** [comment]
Today i set it up manually since i wasn't seeing the config.html either (and it connected to network, but no 
SigGen interface), later i found out you need to send the contents of the data folder to the esp32 flash memory. You can use esp32fs https://github.com/me-no-dev/arduino-esp32fs-plugin/releases/ you add this tool to your arduino ide and then, under tools, you click on "ESP32 Sketch Data Upload" (while having the data folder in the same folder as the SigGen sketch) and it uploads index.html to the esp32, and now you can simply go to 192.168.4.1 no need to use /config or anything and it will show you the interface for both the signal generator and the configuration so you can have it connect to your network!

**Comment by DarS007** [comment]
@Anthocyanina - yes, indeed, the ESP32 Sketch Data upload is mandatory for the web interface to function! This is described in the 'Building' section of the project. So my issue was **not** about _missing_ web interface. In my case the web interface was present. But the Configuration page (the page where you define WiFi parameters) was **not** present when accessed from mobile phone. Everything went to normal when the same web interface was accessed from my notebook. Larger screen made the difference.
So there's a bug in the way the web interface is rendered on mobile screens. The easiest workaround is to use large screens for the initial configuration.
-DarS

**Comment by Anthocyanina** [comment]
@DarS007 oooh, i totally missed that one line! but it sounds like hunabapr also missed it, hopefully they'd have figured it out already, thank you for letting me know that step was already mentioned! and yeah, it's odd, i did everything on my pc and now just checked on my phone and yeah, weird they didn't add the config portion on the mobile version.
